src/DataStructures/Tensor/Tensor.hpp

```diff
@@ -273,6 +273,17 @@ class Tensor<X, Symm, IndexList<Indices...>> {
                                  tmpl::integral_list<UpLo, Indices::ul...>>,
                   "The valences of the generic indices in the expression do "
                   "not match the valences of the indices in the Tensor.");
+    static_assert(
+        std::is_same_v<
```

Comment:
I think this:
```cpp
(... and (N::is_spacetime and Indices::index_type == IndexType::Spacetime))
```
https://en.cppreference.com/w/cpp/language/fold

Reply:
Wow, that syntax (and language feature) is really interesting. I'll try this!